Update hierarchy docs for June 6, 2024 (#1669)
letsencrypt/website: content/en/certificates.md

</details>

<details>
<summary> Delegated OCSP Responder</summary>

This keypair was previously used to sign OCSP responses regarding the status of Let's Encrypt's intermediates on behalf of Let's Encrypt's root, so that the root could remain safely offline. We no longer issue OCSP responses for our intermediates; we instead periodically issue CRLs from our root to convey the revocation status of our intermediates.

* **ISRG Root OCSP X1**
  * Subject: `O = Internet Security Research Group, CN = ISRG Root OCSP X1`
  * Key type: `RSA 2048`
  * Validity: until 2025-06-10
  * Certificate details (signed by ISRG Root X1): [crt.sh](https://crt.sh/?id=2929281974), [der](/certs/isrg-root-ocsp-x1.der), [pem](/certs/isrg-root-ocsp-x1.pem), [txt](/certs/isrg-root-ocsp-x1.txt)
  * Certificate details (signed by ISRG Root X1): [crt.sh](https://crt.sh/?id=142051103) (expired)

</details>
<p><!-- to get the right line spacing after a block element --></p>

# Chains

> When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes with a "chain" of one or more additional certificates. Usually this chain consists of just the Subscriber certificate itself and one intermediate, but it could contain additional intermediates. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.

Reviewer: This still has inconsistent wording as to whether the EE certificate is part of the chain or not. How about (and I'm also sneaking in a Subscriber -> EE tweak):

```suggestion
When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes as part of a "chain" that also includes one or more intermediates. Usually this chain consists of just the end-entity certificate and one intermediate, but it could contain additional intermediates. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.
```

Though having written that, the second sentence now feels pretty redundant with the first; "one or more" implies "could contain additional."

> When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes as part of a "chain" that also includes more or more intermediates. Usually this chain consists of just the end-entity certificate and one intermediate, but it could contain additional intermediates. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.

Reviewer: ```suggestion
When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes as part of a "chain" that also includes one or more intermediates. Usually this chain consists of just the end-entity certificate and one intermediate, but it could contain additional intermediates. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.
```

More is more, but we want "one or more" here. :D

Author: But jacob, it's more _or_ more! So many choices!

linkTitle: Chain of Trust (Root and Intermediate Certificates)
slug: certificates
top_graphic: 5
lastmod: 2024-05-03
show_lastmod: 1
---

<div style="text-align: center">

> _**Note:** This section describes Let's Encrypt's hierarchy as it will be on June 6, 2024._

Reviewer 2: ```suggestion
_**Note:** This section describes Let's Encrypt's hierarchy as it will be starting June 6, 2024._
```

Reviewer: Also this should include a link saying: "To see a description of the hierarchy in use before June 6, 2024, \[see below\](...)"

Or, relatedly, if we wanted to future-proof the tenses (in case we neglect to update the page on June 6), this could say:

This section describes Let's Encrypt's hierarchy after June 6, 2024. For the hierarchy in use before June 6, 2024, \[see below\](...)

Author: Both good ideas. I've rephrased and added a link down to the old section.

</div>

This page describes all of the current and relevant historical Certification Authorities operated by Let's Encrypt. Note that a CA is most correctly thought of as a key and a name: any given CA may be represented by _multiple_ certificates which all contain the same Subject and Public Key Information. In such cases, we have provided the details of all certificates which represent the CA.

[![ISRG Certificate Hierarchy Diagram, as of June 2024](/images/isrg-hierarchy.png)](/images/isrg-hierarchy.png)

# Root CAs

Our root key material is kept safely offline. We issue end-entity certificates to subscribers from the intermediates described in the next section. All root certificate Subjects have a Country field of `C = US`.

> Note that Root CAs don't have expiration dates in quite the same way that other certificates do. Although their self-signed certificates do contain a `notAfter` date, Root Programs and Trust Stores may decide to trust a Root CA beyond that date, or terminate trust in it before that date. As such, the end-of-validity dates given below are approximate, based on current Root Program policies.

Reviewer: Related to this, it's probably useful to include a "generated on" date, since that's what root programs plan to use to remove roots later.

</details>

<details>
<summary> Delegated OCSP Responder</summary>

This keypair was previously used to sign OCSP responses regarding the status of Let's Encrypt's intermediates on behalf of Let's Encrypt's root, so that the root could remain safely offline. We no longer issue OCSP responses for our intermediates; we instead periodically issue CRLs from our root to convey the revocation status of our intermediates.

* **ISRG Root OCSP X1**
  * Subject: `O = Internet Security Research Group, CN = ISRG Root OCSP X1`
  * Key type: `RSA 2048`
  * Validity: until 2025-06-10
  * Certificate details (signed by ISRG Root X1): [crt.sh](https://crt.sh/?id=2929281974), [der](/certs/isrg-root-ocsp-x1.der), [pem](/certs/isrg-root-ocsp-x1.pem), [txt](/certs/isrg-root-ocsp-x1.txt)
  * Certificate details (signed by ISRG Root X1): [crt.sh](https://crt.sh/?id=142051103) (expired)

</details>
<p><!-- to get the right line spacing after a block element --></p>

# Chains

> When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes with a "chain" of one or more intermediates. Usually this chain consists of just the Subscriber certificate itself and the one intermediate which issued it, but it could contain three or more certs, each having issued the previous certificate in the chain. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.

Reviewer: ```suggestion
When an ACME client downloads a newly-issued certificate from Let's Encrypt's ACME API, that certificate comes with a "chain" of one or more certificates. Usually this chain consists of just the Subscriber certificate itself and one intermediate, but it could contain additional intermediates. The idea is that, by presenting this whole chain of certificates to a website visitor's browser, the browser will be able to validate the signatures all the way up to a root that browser trusts without having to download any additional intermediates.
```

Explanation:

- The first and second sentence contradicted each other: chain consists of "certificates", not just "intermediates".
- Better to say "certificates" than "certs".
- As of TLS 1.3 the certificates in the chain are not required to certify the previous ones.
https://datatracker.ietf.org/doc/html/rfc8446#section-4.4.2, see "Note: Prior to TLS 1.3"

show_lastmod: 1
---

> This page describes all of the current and relevant historical Certification Authorities operated by Let's Encrypt. Note that a CA is most correctly thought of as a key and a name: any given CA may be represented by _multiple_ certificates which all contain the same Subject and Public Key Information. In such cases, we have provided the details of all certificates which represent the CA.

Reviewer 3: Update the lastmod date please.

* **ISRG Root X1**
  * Subject: `O = Internet Security Research Group, CN = ISRG Root X1`
  * Key type: `RSA 4096`
  * Validity: until 2030-06-04
  * CA details: [crt.sh](https://crt.sh/?caid=7394), [issued certs](https://crt.sh/?Identity=%25&iCAID=7394)
  * Certificate details (self-signed): [crt.sh](https://crt.sh/?id=9314791), [der](/certs/isrgrootx1.der), [pem](/certs/isrgrootx1.pem), [txt](/certs/isrgrootx1.txt)
  * Certificate details (cross-signed by DST Root CA X3): [crt.sh](https://crt.sh/?id=3958242236), [der](/certs/isrg-root-x1-cross-signed.der), [pem](/certs/isrg-root-x1-cross-signed.pem), [txt](/certs/isrg-root-x1-cross-signed.txt) (retired)
  * Test websites: [valid](https://valid-isrgrootx1.letsencrypt.org/), [revoked](https://revoked-isrgrootx1.letsencrypt.org/), [expired](https://expired-isrgrootx1.letsencrypt.org/)
* **ISRG Root X2**
  * Subject: `O = Internet Security Research Group, CN = ISRG Root X2`
  * Key type: `ECDSA P-384`
  * Validity: until 2035-09-04

>   * CA details: [crt.sh](https://crt.sh/?caid=7394), [issued certs](https://crt.sh/?Identity=%25&iCAID=7394)

Reviewer 3: The old link was a copy/paste of IRSG Root X1.
```suggestion
  * CA details: [crt.sh](https://crt.sh/?caid=183269), [issued certs](https://crt.sh/?Identity=%25&iCAID=183269)
```
